src/overlays/actors/ovl_En_Dg/z_en_dg.c

```diff
@@ -92,114 +147,1098 @@ static DamageTable D_8098C33C = {
+                    this->unk_28C = 3;
+                    func_80989ADC(this, globalCtx);
+                }
+                break;
+
+            case PLAYER_FORM_DEKU:
+                this->unk_280 &= ~2;
+                if ((this->unk_28C != 6) && (player->actor.speedXZ > 1.0f)) {
+                    this->unk_28C = 6;
+                    func_80989140(&this->skelAnime, D_8098C35C, 2);
+                    this->actionFunc = func_8098A938;
+                }
+
+                if ((this->unk_28C != 7) && (this->unk_28C != 6)) {
+                    this->unk_28C = 7;
+                    func_80989ADC(this, globalCtx);
+                }
+                break;
+        }
+    }
+}
+
+void func_8098A468(EnDg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = -3.0f;
+    func_80989674(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    if (player->transformation == PLAYER_FORM_HUMAN) {
+        func_80989E18(this, globalCtx);
+    }
+
+    func_8098A064(this, globalCtx);
+    func_80989974(this);
+    if (!(this->actor.bgCheckFlags & 1)) {
+        this->actionFunc = func_8098AF98;
+    }
+
+    if (DECR(this->unk_282) == 0) {
+        this->unk_282 = Rand_S16Offset(20, 20);
+        func_80989140(&this->skelAnime, D_8098C35C, 3);
+        this->actionFunc = func_8098A55C;
+    }
+}
+
+void func_8098A55C(EnDg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (player->transformation == PLAYER_FORM_HUMAN) {
+        func_80989E18(this, globalCtx);
+    }
+
+    func_809899C8(this, 13.0f);
+    func_809899C8(this, 19.0f);
+    if (!(this->actor.bgCheckFlags & 1)) {
+        this->actionFunc = func_8098AF98;
+    }
+
+    if (DECR(this->unk_282) == 0) {
+        this->unk_282 = Rand_S16Offset(60, 60);
+        func_80989ADC(this, globalCtx);
+    }
+}
+
+void func_8098A618(EnDg* this, GlobalContext* globalCtx) {
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = -3.0f;
+
+    if (DECR(this->unk_282) == 0) {
+        this->unk_282 = 50;
+        func_80989140(&this->skelAnime, D_8098C35C, 2);
+        this->actionFunc = func_8098A70C;
+    } else {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 0x3E8, 1);
+        if (this->actor.bgCheckFlags & 8) {
+            this->actor.shape.rot.y = this->actor.wallYaw;
+        }
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        Math_ApproachF(&this->actor.speedXZ, -1.5f, 0.2f, 1.0f);
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    }
+
+    func_80989974(this);
+    func_80989A9C(this, 0);
+}
+
+void func_8098A70C(EnDg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 phi_a1;
+
+    if (this->actor.xzDistToPlayer < 250.0f) {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer * -1, 4, 0xC00);
```

Comment:
```suggestion
        Math_ApproachS(&this->actor.shape.rot.y, -this->actor.yawTowardsPlayer, 4, 0xC00);
```
Would this still match?